// terriajs/lib/ReactViews/PropertySearchWindow/Tabs/DataCatalogTab.jsx
import React from 'react';

import createReactClass from 'create-react-class';

import PropTypes from 'prop-types';

import defined from 'terriajs-cesium/Source/Core/defined';
import DataCatalog from '../../SearchResultDataCatalog/DataCatalog.jsx';
import DataCatalogMember from '../../SearchResultDataCatalog/DataCatalogMember.jsx';
import DataPreview from '../../Preview/DataPreview.jsx';
import ObserveModelMixin from '../../ObserveModelMixin';
import SearchHeader from '../../Search/SearchHeader.jsx';
import SearchBox from '../../Search/SearchBox.jsx';
import imgCardMelli from "../../../../wwwroot/images/National_Card_1.jpg";

import Styles from './data-catalog-tab.scss';

// The DataCatalog Tab
const DataCatalogTab = createReactClass({
    displayName: 'DataCatalogTab',
    mixins: [ObserveModelMixin],
    searchByNameResult : "",
    propTypes: {
        terria: PropTypes.object,
        viewState: PropTypes.object
    },

    changeSearchText(newText) {
        this.props.viewState.searchState.catalogSearchText = newText;
    },

    search() {
        this.props.viewState.searchState.searchCatalog();
    },

    render() {
        const terria = this.props.terria;
        return (
            <div className={Styles.root}>
                <div className={Styles.dataExplorer}>
                    <SearchBox searchText={this.props.viewState.searchState.catalogSearchText}
                               onSearchTextChanged={this.changeSearchText}
                               onDoSearch={this.search} />
                      <h3 className={Styles.h3}>
                       <img src={imgCardMelli} className={Styles.img} hidden={!this.props.viewState.searchPropertyByMelliCode && !this.props.viewState.searchPropertyByName} />
                        {this.props.viewState.searchByNameResult}</h3>
                    <DataCatalog terria={this.props.terria}
                                 viewState={this.props.viewState} />
                                
                </div>
                <DataPreview terria={terria}
                             viewState={this.props.viewState}
                             previewed={this.props.viewState.previewedItem}
                />
            </div>
        );
    },

    renderDataCatalog() {
        const terria = this.props.terria;
        const searchState = this.props.viewState.searchState;
        const isSearching = searchState.catalogSearchText.length > 0;
        const items = (
            isSearching ?
                searchState.catalogSearchProvider.searchResults.map(result => result.catalogItem) :
                terria.catalog.group.items
        ).filter(defined);

        return (
            <ul className={Styles.dataCatalog}>
                <If condition={isSearching}>
                    <label className={Styles.label}>Search results</label>
                    <SearchHeader searchProvider={searchState.catalogSearchProvider}
                                  isWaitingForSearchToStart={searchState.isWaitingToStartCatalogSearch}/>
                </If>
                <For each="item" of={items}>
                    {item !== this.props.terria.catalog.userAddedDataGroup &&
                        <DataCatalogMember viewState={this.props.viewState}
                                           member={item}
                                           manageIsOpenLocally={isSearching}
                                           key={item.uniqueId}
                    />}
                </For>
            </ul>
        );
    },
});

module.exports = DataCatalogTab;
