// terriajs/lib/ReactViews/PropertySearchWindow/Tabs.jsx
import React from 'react';
import createReactClass from 'create-react-class';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import DataCatalogTab from './Tabs/DataCatalogTab.jsx';
import MyDataTab from './Tabs/MyDataTab/MyDataTab.jsx';
import ObserveModelMixin from '../ObserveModelMixin';

import Styles from './tabs.scss';
const Tabs = createReactClass({
    displayName: 'Tabs',
    mixins: [ObserveModelMixin],

    propTypes: {
        terria: PropTypes.object.isRequired,
        viewState: PropTypes.object.isRequired,
        tabs: PropTypes.array
    },
    isHidden(v) {
        //if (v = 0) { return true; }
        return true;
    },
    getInitialState() {
        // This can be passed in as prop
        return {
            tabs: this.props.tabs || [
                {
                    name: 'جستجواملاک',
                    title: 'my-data',
                    panel: <MyDataTab terria={this.props.terria}
                                      viewState={this.props.viewState}
                    />
                },
                {
                    name: 'نتیجه جستجو',
                    title: 'data-catalog',
                    panel: <DataCatalogTab terria={this.props.terria}
                                           viewState={this.props.viewState}
                    />
                }
            ]
        };
    },

    activateTab(i) {
        this.props.viewState.modalPropertyTabIndex = i;
    },
    isTabVisible(item) {
        if (item.title == "data-catalog") {
            if (this.props.viewState.isResultVisible) { 
            this.activateTab.bind(this,1);
        }
            return this.props.viewState.isResultVisible;
        }
        return true;
    },
    render() {
        const item = this.state.tabs[this.props.viewState.modalPropertyTabIndex];
        return (
            <div className={Styles.tabs}>
                <ul className={Styles.tabList} role="tablist">
                    <For each="item" index="i" of={this.state.tabs}>
                        <If condition={this.isTabVisible(item)}>
                        <li key={i}
                            id={'tablist--' + item.title}
                            className={Styles.tabListItem}
                            role="tab"
                            aria-controls={'panel--' + item.title}
                            aria-selected={this.props.viewState.modalPropertyTabIndex === i}>
                            <button type='button'
                                    onClick={this.activateTab.bind(this, i)}
                                    className={classNames(Styles.btnTab, {[Styles.btnSelected]: this.props.viewState.modalPropertyTabIndex === i})}>
                                {item.name}
                            </button>
                        </li>
                         </If>
                    </For>
                </ul>

                <section
                    key={item.title}
                    id={'panel--' + item.title}
                    className={classNames(Styles.tabPanel, Styles.isActive)}
                    aria-labelledby={'tablist--' + item.title}
                    role='tabpanel' tabIndex='0'>
                    <div className={Styles.panelContent}>
                        {item.panel}
                    </div>
                </section>
            </div>
        );
    },
});

module.exports = Tabs;
