// terriajs/lib/ReactViews/PropertySearchWindow/Tabs/MyDataTab/MyDataTab.jsx
import React from 'react';

import createReactClass from 'create-react-class';

import PropTypes from 'prop-types';

import DataCatalogGroup from '../../../DataCatalog/DataCatalogGroup.jsx';
import DataPreview from '../../../Preview/DataPreview.jsx';
import AddData from './AddData.jsx';
import ObserveModelMixin from '../../../ObserveModelMixin';
import Dropdown from '../../../../../../react-dropdown';
//import Dropdown2 from '../../../Generic/Dropdown';
import Styles from './my-data-tab.scss';
import '!!style-loader!css-loader?sourceMap!../../../../../../react-dropdown/style.css';

const options = [
    { value: 'name', label: 'جستجو بر اساس نام و نام خانوادگي' },
    { value: 'melliCode', label: 'جستجو بر اساس کد ملي' },
    { value: 'address', label: 'جستجو بر اساس آدرس' },
    { value: 'postCode', label: ' جستجو بر اساس کد پستی' }
]
const optionsMelicode = [
    { value :'0', label :'لطفا کد ملی موردنظر را انتخاب کنید'},
    { value: '1', label: '12323123'},
    { value:'2', label: '23423424'}
]

//const defaultOption = options[0]
const defaultOptionMeliCode = optionsMelicode[0]

// My data tab include Add data section and preview section
const MyDataTab = createReactClass({
    getInitialState: function () {
        return {
            firstNameValue: 'داود',
            surnameValue: 'شجاعی',
            melliCodeValue: '0534886574',
            cityValue: 'تهران',
            streetValue: 'بهشتی',
            streetNumberValue: '۲',
            postcodeValue: '157714671',
            defaultoption: options[0],

        };
       
    },
    displayName: 'PropertySearchTab',
    mixins: [ObserveModelMixin],

    propTypes: {
        terria: PropTypes.object,
        viewState: PropTypes.object
    },
// Added by Farzad
    onSelect(val) {
        this.state.defaultoption = val;
         if (val.value === 'name' ){
            this.props.viewState.searchPropertyByName = true;   
            this.props.viewState.searchPropertyByAddress = false;
            this.props.viewState.searchPropertyByMelliCode = false;
            this.props.viewState.searchPropertyByPostCode = false;  
        } else if (val.value === 'address'){
            this.props.viewState.searchPropertyByName = false;   
            this.props.viewState.searchPropertyByAddress = true;
            this.props.viewState.searchPropertyByMelliCode = false;
            this.props.viewState.searchPropertyByPostCode = false;            
        } else if (val.value === 'melliCode') {
            this.props.viewState.searchPropertyByName = false;   
            this.props.viewState.searchPropertyByAddress = false;
            this.props.viewState.searchPropertyByMelliCode = true;
            this.props.viewState.searchPropertyByPostCode = false;              
        } else if (val.value === 'postCode') {
            this.props.viewState.searchPropertyByName = false;   
            this.props.viewState.searchPropertyByAddress = false;
            this.props.viewState.searchPropertyByMelliCode = false;
            this.props.viewState.searchPropertyByPostCode = true;          
        }
    },
    onsubmitMellicode(val) {
        if (val.value != "0")
        {
            this.props.viewState.searchByNameResult = this.state.firstNameValue + '  ' + this.state.surnameValue + ' کدملی:  '  +val.label ;
            this.props.viewState.isResultVisible = true;
            document.getElementById("tablist--data-catalog").getElementsByTagName("button")[0].click();
        }
    },
    submit(e) {
        switch (e.target.name) {
            case 'searchByNameBtn':
                setTimeout(function () { this.props.viewState.isSearchByNameClicked = true; }.bind(this), 2000);
                break;
            case 'searchByMelliCodeBtn':
                this.props.viewState.searchByNameResult = this.state.firstNameValue + '  ' + this.state.surnameValue + ' کدملی:  ' + this.state.melliCodeValue;
                this.props.viewState.isResultVisible = true;
                setTimeout(function () { document.getElementById("tablist--data-catalog").getElementsByTagName("button")[0].click() }.bind(this),0.5);
                break;
            case 'searchByAddressBtn':
                this.props.viewState.searchByNameResult =  this.state.cityValue + ' خیابان ' + this.state.streetValue + ' پلاک  ' + this.state.streetNumberValue;
                this.props.viewState.isResultVisible = true;
                setTimeout(function () { document.getElementById("tablist--data-catalog").getElementsByTagName("button")[0].click() }.bind(this), 0.5);
                 break;
            case 'searchBypostcodeBtn':
                console.log("dff");
                this.props.viewState.searchByNameResult = 'تهران خیابان بهشتی پلاک 2 ' + ' کد پستی' + this.state.postcodeValue;
                this.props.viewState.isResultVisible = true;
                setTimeout(function () { document.getElementById("tablist--data-catalog").getElementsByTagName("button")[0].click() }.bind(this), 0.5);
                break;
            default:
            //  code block
        }
        
    },
    onBackButtonClicked() {
        this.props.viewState.myDataIsUploadView = false;
    },

    onAddMoreDataButtonClicked() {
        this.props.viewState.myDataIsUploadView = true;
    },

    hasUserAddedData() {
        return this.props.terria.catalog.userAddedDataGroup.items.length > 0;
    },
    
    handleChange(e) {
        switch (e.target.name ) {
            case 'nameField':
                this.setState({ firstNameValue: e.target.value });
                break;
            case 'surnameField':
                this.setState({ surnameValue: e.target.value });
                break;
            case 'melliCodeField':
                this.setState({ melliCodeValue: e.target.value });
                break;
            case 'cityField':
                this.setState({ cityValue: e.target.value });
                break;
            case 'streetNameField':
                this.setState({ streetValue: e.target.value });
                break;
            case 'streetNumberField':
                this.setState({ streetNumberValue: e.target.value });
                break;
            case 'postcodeField':
                this.setState({ postcodeValue: e.target.value });
                break;
            default:
              //  code block
        }
           
       

    },
    

    render() {
       
        return (
            <div className={Styles.root}>
                <div className={Styles.leftCol}>
                    <If condition={this.props.viewState.myDataIsUploadView}>
                        <div className={Styles.addedData}>
                            <If condition={this.hasUserAddedData()}>
                                <button type='button'
                                        onClick={this.onBackButtonClicked}
                                        className={Styles.btnBackToMyData}>
                                    Back
                                </button>
                            </If>
                            <h3 className={Styles.h3}>جستجوی املاک</h3>
                            <Dropdown id='searchProperySelectTypeID' options={options} onChange={this.onSelect} value={this.state.defaultoption} placeholder="Select an option" />
                           
                            {/*<AddData terria={this.props.terria}
                                     viewState={this.props.viewState}/>*/}

                            <If condition={this.props.viewState.searchPropertyByAddress}>
                                <div>                       
                                    <label className={Styles.label} htmlFor="cityField">شهر </label>
                                    <input className={Styles.field} type='text' name='cityField' value={this.state.cityValue} onChange={this.handleChange} />
                                    <label className={Styles.label} htmlFor="streetNameField">خیابان </label>
                                    <input className={Styles.field} type='text' name='streetNameField' value={this.state.streetValue} defaultValue="بهشتی" onChange={this.handleChange} />
                                    <label className={Styles.label} htmlFor="streetNumberField">پلاک </label>
                                    <input className={Styles.field} type='text' name='streetNumberField' value={this.state.streetNumberValue} defaultValue="۲" onChange={this.handleChange} />                                                 
                                    <button type='button' className={Styles.btn} onClick={this.submit} name="searchByAddressBtn">جستجو</button>

                                </div>
                            </If>
                            <If condition={this.props.viewState.searchPropertyByName}>
                                <div>
                                    <label className={Styles.label} htmlFor="nameField">نام </label>
                                    <input className={Styles.field} type='text' value={this.state.firstNameValue} defaultValue='داود' name='nameField' onChange={this.handleChange} />
                                    <label className={Styles.label} htmlFor="surnameField" value={this.state.surnameValue} onChange={this.handleChange} > نام خانوادگی </label>
                                    <input className={Styles.field} type='text' defaultValue='شجاعی' name='sureNameField' onChange={this.changeRangeMin} />
                                    <button type='button' className={Styles.btn} onClick={this.submit} name="searchByNameBtn">جستجو</button>
                                    <If condition={this.props.viewState.isSearchByNameClicked}>
                                    <p >
                                            لطفا یکی را انتخاب کنید
                                   
                                     <Dropdown options={optionsMelicode} onChange={this.onsubmitMellicode} value={optionsMelicode[0]}  placeholder="Select an option" />
                                    </p>
                                    </If>
                                </div>
                            </If>          
                            <If condition={this.props.viewState.searchPropertyByMelliCode}>
                                <div>
                                    <label className={Styles.label} htmlFor="melliCodeField"  > کد ملی </label>
                                    <input className={Styles.field} type='text' defaultValue="0534886574" name='melliCodeField' onChange={this.handleChange} />
 
                                    <button type='button' className={Styles.btn} onClick={this.submit} name="searchByMelliCodeBtn">جستجو</button>
                                   
                                </div>
                            </If>      
                            <If condition={this.props.viewState.searchPropertyByPostCode}>
                                <div>
                                    <label className={Styles.label} htmlFor="postcodeField">کد پستی </label>
                                    <input className={Styles.field} type='text' defaultValue="157714671" name='postcodeField' onChange={this.handleChange} />
 
                                    <button type='button' className={Styles.btn} onClick={this.submit} name="searchBypostcodeBtn">جستجو</button>
                                    
                                </div>
                            </If>                                                                                 
                        </div>
                    </If>
                    <If condition={this.hasUserAddedData()}>
                        <div className={Styles.addedData}>
                            <p className={Styles.explanation}>
                                Data added in this way is not saved or made visible to others unless you explicitly share
                                it by using the Share panel.
                            </p>
                            <ul className={Styles.dataCatalog}>
                                <DataCatalogGroup group={this.props.terria.catalog.userAddedDataGroup}
                                                  viewState={this.props.viewState}/>
                            </ul>
                            <button type='button'
                                    onClick={this.onAddMoreDataButtonClicked}
                                    className={Styles.btnAddMoreData}>
                                Add more data
                            </button>
                        </div>
                    </If>
                </div>
                <DataPreview terria={this.props.terria}
                             viewState={this.props.viewState}
                             previewed={this.props.viewState.userDataPreviewedItem}
                />
            </div>
        );
    },
});

module.exports = MyDataTab;
