// terriajs/lib/ReactViews/PropertySearchWindow/Tabs/MyDataTab/MyDataTab.jsx
import React from 'react';

import createReactClass from 'create-react-class';

import PropTypes from 'prop-types';

import DataCatalogGroup from '../../../DataCatalog/DataCatalogGroup.jsx';
import DataPreview from '../../../Preview/DataPreview.jsx';
import AddData from './AddData.jsx';
import ObserveModelMixin from '../../../ObserveModelMixin';
import Dropdown from '../../../../../../react-dropdown';
import Dropdown2 from '../../../Generic/Dropdown';
import Styles from './my-data-tab.scss';
import '!!style-loader!css-loader?sourceMap!../../../../../../react-dropdown/style.css';

const options = [
    { value: 'name', label: 'جستجو بر اساس نام و نام خانوادگي' },
    { value: 'melliCode', label: 'جستجو بر اساس کد ملي' },
    { value: 'address', label: 'جستجو بر اساس آدرس' },
    { value: 'postCode', label: ' جستجو بر اساس کد پستی' }
]
const optionsMelicode = [
    { value: '1', label: '12323123'},
    { value:'2', label: '23423424'}
]

const defaultOption = options[0]
const defaultOptionMeliCode = optionsMelicode[0]

// My data tab include Add data section and preview section
const MyDataTab = createReactClass({
    displayName: 'PropertySearchTab',
    mixins: [ObserveModelMixin],

    propTypes: {
        terria: PropTypes.object,
        viewState: PropTypes.object
    },
// Added by Farzad
    onSelect(val) {
      console.log(val.value);
         if (val.value === 'name' ){
            this.props.viewState.searchPropertyByName = true;   
            this.props.viewState.searchPropertyByAddress = false;
            this.props.viewState.searchPropertyByMelliCode = false;
            this.props.viewState.searchPropertyByPostCode = false;  
        } else if (val.value === 'address'){
            this.props.viewState.searchPropertyByName = false;   
            this.props.viewState.searchPropertyByAddress = true;
            this.props.viewState.searchPropertyByMelliCode = false;
            this.props.viewState.searchPropertyByPostCode = false;            
        } else if (val.value === 'melliCode') {
            this.props.viewState.searchPropertyByName = false;   
            this.props.viewState.searchPropertyByAddress = false;
            this.props.viewState.searchPropertyByMelliCode = true;
            this.props.viewState.searchPropertyByPostCode = false;              
        } else if (val.value === 'postCode') {
            this.props.viewState.searchPropertyByName = false;   
            this.props.viewState.searchPropertyByAddress = false;
            this.props.viewState.searchPropertyByMelliCode = false;
            this.props.viewState.searchPropertyByPostCode = true;          
        }
    },
    submitMellicode() {
        
    },
    submitName(){
        setTimeout(function () { this.props.viewState.isSearchByNameClicked = true; }.bind(this), 2000);
        
    },
    onBackButtonClicked() {
        this.props.viewState.myDataIsUploadView = false;
    },

    onAddMoreDataButtonClicked() {
        this.props.viewState.myDataIsUploadView = true;
    },

    hasUserAddedData() {
        return this.props.terria.catalog.userAddedDataGroup.items.length > 0;
    },

    render() {
       
        return (
            <div className={Styles.root}>
                <div className={Styles.leftCol}>
                    <If condition={this.props.viewState.myDataIsUploadView}>
                        <div className={Styles.addedData}>
                            <If condition={this.hasUserAddedData()}>
                                <button type='button'
                                        onClick={this.onBackButtonClicked}
                                        className={Styles.btnBackToMyData}>
                                    Back
                                </button>
                            </If>
                            <h3 className={Styles.h3}>جستجوی املاک</h3>
                            <Dropdown id='searchProperySelectTypeID' options={options} onChange={this.onSelect} value={options[0]} placeholder="Select an option" />
                            
                            {/*<AddData terria={this.props.terria}
                                     viewState={this.props.viewState}/>*/}

                            <If condition={this.props.viewState.searchPropertyByAddress}>
                                <div>                       
                                    <label className={Styles.label} htmlFor="cityField">شهر </label>
                                    <input className={Styles.field} type='text' name='cityField'  defaultValue="تهران" onChange={this.changeRangeMin} />
                                    <label className={Styles.label} htmlFor="streetNameField">خیابان </label>
                                    <input className={Styles.field} type='text' name='streetNameField' defaultValue="بهشتی" onChange={this.changeRangeMin} />
                                    <label className={Styles.label} htmlFor="streetNumberField">پلاک </label>
                                    <input className={Styles.field} type='text' name='streetNumberField'  defaultValue="۲" onChange={this.changeRangeMin} />                                                 
                                    <button type='button' className={Styles.btn} onClick={this.submit}>جستجو</button>

                                </div>
                            </If>
                            <If condition={this.props.viewState.searchPropertyByName}>
                                <div>
                                    <label className={Styles.label} htmlFor="nameField">نام </label>
                                    <input className={Styles.field} type='text' defaultValue='داود' name='nameField' onChange={this.changeRangeMin} />
                                    <label className={Styles.label} htmlFor="sureNameField">نام خانوادگی </label>
                                    <input className={Styles.field} type='text' defaultValue='شجاعی' name='sureNameField' onChange={this.changeRangeMin} />
                                    <button type='button' className={Styles.btn} onClick={this.submitName}>جستجو</button>
                                    <If condition={this.props.viewState.isSearchByNameClicked}>
                                    <p >
                                            لطفا یکی را انتخاب کنید
                                   
                                     <Dropdown  options={optionsMelicode} onChange={this.onSelect} value={optionsMelicode[0]}  placeholder="Select an option" />
                                    </p>
                                    </If>
                                </div>
                            </If>          
                            <If condition={this.props.viewState.searchPropertyByMelliCode}>
                                <div>
                                    <label className={Styles.label} htmlFor="melliCodeField">کد ملی </label>
                                    <input className={Styles.field} type='text' defaultValue="0534886574" name='melliCodeField' onChange={this.changeRangeMin} />
 
                                    <button type='button' className={Styles.btn} onClick={this.submitMelliCode}>جستجو</button>
                                   
                                </div>
                            </If>      
                            <If condition={this.props.viewState.searchPropertyByPostCode}>
                                <div>
                                    <label className={Styles.label} htmlFor="melliCodeField">کد پستی </label>
                                    <input className={Styles.field} type='text' defaultValue="157714671" name='melliCodeField' onChange={this.changeRangeMin} />
 
                                    <button type='button' className={Styles.btn} onClick={this.submit}>جستجو</button>
                                    
                                </div>
                            </If>                                                                                 
                        </div>
                    </If>
                    <If condition={this.hasUserAddedData()}>
                        <div className={Styles.addedData}>
                            <p className={Styles.explanation}>
                                Data added in this way is not saved or made visible to others unless you explicitly share
                                it by using the Share panel.
                            </p>
                            <ul className={Styles.dataCatalog}>
                                <DataCatalogGroup group={this.props.terria.catalog.userAddedDataGroup}
                                                  viewState={this.props.viewState}/>
                            </ul>
                            <button type='button'
                                    onClick={this.onAddMoreDataButtonClicked}
                                    className={Styles.btnAddMoreData}>
                                Add more data
                            </button>
                        </div>
                    </If>
                </div>
                <DataPreview terria={this.props.terria}
                             viewState={this.props.viewState}
                             previewed={this.props.viewState.userDataPreviewedItem}
                />
            </div>
        );
    },
});

module.exports = MyDataTab;
