// terriajs/lib/ReactViews/PropertySearchWindow/Tabs/MyDataTab/MyDataTab.jsx
import React from 'react';

import createReactClass from 'create-react-class';

import PropTypes from 'prop-types';

import DataCatalogGroup from '../../../DataCatalog/DataCatalogGroup.jsx';
import DataPreview from '../../../Preview/DataPreview.jsx';
import AddData from './AddData.jsx';
import ObserveModelMixin from '../../../ObserveModelMixin';
import Dropdown from '../../../../../../react-dropdown';
//import Dropdown2 from '../../../Generic/Dropdown';
import Styles from './my-data-tab.scss';
import '!!style-loader!css-loader?sourceMap!../../../../../../react-dropdown/style.css';
import img from "../../../../../wwwroot/images/ajax-loader.gif";

const options = [
    { value: 'name', label: 'جستجو بر اساس نام و نام خانوادگي' },
    { value: 'melliCode', label: 'جستجو بر اساس کد ملي' },
    { value: 'address', label: ' GNAF جستجو بر اساس آدرس' },
    { value: 'postCode', label: ' جستجو بر اساس کد پستی' }
]
const optionsMelicode = [
    { value: '0', label: 'لطفا کد ملی موردنظر را انتخاب کنید' },
    { value: '1', label: '0534886574' },
    { value: '2', label: '0348657423' }
]

//const defaultOption = options[0]
const defaultOptionMeliCode = optionsMelicode[0]

// My data tab include Add data section and preview section
const MyDataTab = createReactClass({
    getInitialState: function () {
        return {
            firstNameValue: 'علی ',
            surnameValue: 'احمدی ',
            melliCodeValue: '0534886574',
            cityValue: 'تهران',
            streetValue: 'خیابان قاضی زاده / خیابان درویش ',
            suburbValue: 'خزانه / 38137',
            streetNumberValue: ' 1 / 135',
            postcodeValue: '3813783188',
            defaultoption: options[0],
            initoption: options[0]

        };

    },
    displayName: 'PropertySearchTab',
    mixins: [ObserveModelMixin],

    propTypes: {
        terria: PropTypes.object,
        viewState: PropTypes.object
    },
    // Added by Farzad
    onSelect(val) {
        this.props.viewState.isSearchByNameClicked = false;
        this.state.defaultoption = val;
        if (val.value === 'name') {
            this.props.viewState.searchPropertyByName = true;
            this.props.viewState.searchPropertyByAddress = false;
            this.props.viewState.searchPropertyByMelliCode = false;
            this.props.viewState.searchPropertyByPostCode = false;
        } else if (val.value === 'address') {
            this.props.viewState.searchPropertyByName = false;
            this.props.viewState.searchPropertyByAddress = true;
            this.props.viewState.searchPropertyByMelliCode = false;
            this.props.viewState.searchPropertyByPostCode = false;
        } else if (val.value === 'melliCode') {
            this.props.viewState.searchPropertyByName = false;
            this.props.viewState.searchPropertyByAddress = false;
            this.props.viewState.searchPropertyByMelliCode = true;
            this.props.viewState.searchPropertyByPostCode = false;
        } else if (val.value === 'postCode') {
            this.props.viewState.searchPropertyByName = false;
            this.props.viewState.searchPropertyByAddress = false;
            this.props.viewState.searchPropertyByMelliCode = false;
            this.props.viewState.searchPropertyByPostCode = true;
        }
    },
    onsubmitMellicode(val) {
        if (val.value != "0") {
            this.props.viewState.searchByNameResult = this.state.firstNameValue + '  ' + this.state.surnameValue + ' کدملی:  ' + val.label;
            this.props.viewState.isResultVisible = true;
            document.getElementById("tablist--data-catalog").getElementsByTagName("button")[0].click();
        }
    },
    submit(e) {



        switch (e.target.name) {
            case 'searchByNameBtn':
                this.props.viewState.isLoadingSearchResult = true;
                setTimeout(function () { this.props.viewState.isLoadingSearchResult = false; this.props.viewState.isSearchByNameClicked = true; }.bind(this), 1000);
                break;
            case 'searchByMelliCodeBtn':
                this.props.viewState.isLoadingSearchResult = true;
                this.props.viewState.searchByNameResult = this.state.firstNameValue + '  ' + this.state.surnameValue + ' کدملی:  ' + this.state.melliCodeValue;

                setTimeout(function () { this.props.viewState.isLoadingSearchResult = false; this.props.viewState.isResultVisible = true; document.getElementById("tablist--data-catalog").getElementsByTagName("button")[0].click() }.bind(this), 1000);
                break;
            case 'searchByAddressBtn':
                this.props.viewState.isLoadingSearchResultPost = true;
                this.props.viewState.searchByNameResult = ((this.state.cityValue + '، ' + this.state.suburbValue + '، ' + this.state.streetValue + '/ پلاک' + (this.state.streetNumberValue).replace("/ ", "/واحد ")).replace(" /", "،")).replace(" /", "،").replace(" /", "،").replace(" /", "، ");

                setTimeout(function () { this.props.viewState.isLoadingSearchResultPost = false; this.props.viewState.isResultVisible = true; document.getElementById("tablist--data-catalog").getElementsByTagName("button")[0].click() }.bind(this), 1000);
                break;
            case 'searchBypostcodeBtn':
                this.props.viewState.isLoadingSearchResultPost = true;
                this.props.viewState.searchByNameResult = 'تهران، خزانه، 38137، خیابان قاضی زاده، خیابان درویش، پلاک 1، واحد 135';

                setTimeout(function () { this.props.viewState.isLoadingSearchResultPost = false; this.props.viewState.isResultVisible = true; document.getElementById("tablist--data-catalog").getElementsByTagName("button")[0].click() }.bind(this), 1000);
                break;
            default:
            //  code block
        }


    },
    onBackButtonClicked() {
        this.props.viewState.myDataIsUploadView = false;
    },

    onAddMoreDataButtonClicked() {
        this.props.viewState.myDataIsUploadView = true;
    },

    hasUserAddedData() {
        return this.props.terria.catalog.userAddedDataGroup.items.length > 0;
    },

    handleChange(e) {
        switch (e.target.name) {
            case 'nameField':
                this.setState({ firstNameValue: e.target.value });
                break;
            case 'surnameField':
                this.setState({ surnameValue: e.target.value });
                break;
            case 'melliCodeField':
                this.setState({ melliCodeValue: e.target.value });
                break;
            case 'cityField':
                this.setState({ cityValue: e.target.value });
                break;
            case 'streetNameField':
                this.setState({ streetValue: e.target.value });
                break;
            case 'suburbNameField':
                this.setState({ suburbValue: e.target.value });
                break;
            case 'streetNumberField':
                this.setState({ streetNumberValue: e.target.value });
                break;
            case 'postcodeField':
                this.setState({ postcodeValue: e.target.value });
                break;
            default:
            //  code block
        }



    },


    render() {
        const ajaxloaderpath = "../../../../../wwwroot/images/ajax-loader.gif";
        return (
            <div className={Styles.root}>
                <div className={Styles.leftCol}>
                    <If condition={this.props.viewState.myDataIsUploadView}>
                        <div className={Styles.addedData}>
                            <If condition={this.hasUserAddedData()}>
                                <button type='button'
                                    onClick={this.onBackButtonClicked}
                                    className={Styles.btnBackToMyData}>
                                    Back
                                </button>
                            </If>
                            <h3 className={Styles.h3}>جستجوی املاک</h3>
                            <Dropdown id='searchProperySelectTypeID' options={options} onChange={this.onSelect} value={this.state.defaultoption} placeholder="Select an option" />

                            {/*<AddData terria={this.props.terria}
                                     viewState={this.props.viewState}/>*/}

                            <If condition={this.props.viewState.searchPropertyByAddress}>
                                <div>
                                    <label className={Styles.label} htmlFor="cityField">شهر </label>
                                    <input className={Styles.field} type='text' name='cityField' value={this.state.cityValue} onChange={this.handleChange} />
                                    <label className={Styles.label} htmlFor="suburbNameField">نام محله / کدپستي ۵ رقمي    </label>
                                    <input className={Styles.field} type='text' name='suburbNameField' value={this.state.suburbValue} defaultValue="بهشتی" onChange={this.handleChange} />
                                    <label className={Styles.label} htmlFor="streeNameField">نوع و نام معبر ماقبل آخر / نوع و نام معبرآخر </label>
                                    <input className={Styles.field} type='text' name='streetNameField' value={this.state.streetValue} defaultValue="بهشتی" onChange={this.handleChange} />
                                    <label className={Styles.label} htmlFor="streetNumberField">پلاک / طبقه و واحد </label>
                                    <input className={Styles.field} type='text' name='streetNumberField' value={this.state.streetNumberValue} defaultValue="۲" onChange={this.handleChange} />
                                    <button type='button' className={Styles.btn} onClick={this.submit} name="searchByAddressBtn">جستجو</button>

                                </div>
                            </If>
                            <If condition={this.props.viewState.searchPropertyByName}>
                                <div>
                                    <label className={Styles.label} htmlFor="nameField">نام </label>
                                    <input className={Styles.field} type='text' value={this.state.firstNameValue} defaultValue='علی ' name='nameField' onChange={this.handleChange} />
                                    <label className={Styles.label} htmlFor="surnameField" value={this.state.surnameValue} onChange={this.handleChange} > نام خانوادگی </label>
                                    <input className={Styles.field} type='text' defaultValue='احمدی ' name='sureNameField' onChange={this.changeRangeMin} />
                                    <button type='button' className={Styles.btn} onClick={this.submit} name="searchByNameBtn">جستجو</button>
                                    <If condition={this.props.viewState.isSearchByNameClicked}>
                                        <p className={Styles.pMellicode}>
                                            لطفا یکی از موارد ذیل را انتخاب کنید

                                     <Dropdown options={optionsMelicode} onChange={this.onsubmitMellicode} value={optionsMelicode[0]} placeholder="Select an option" />
                                        </p>
                                    </If>
                                </div>
                            </If>

                            <If condition={this.props.viewState.searchPropertyByMelliCode}>
                                <div>
                                    <label className={Styles.label} htmlFor="melliCodeField"  > کد ملی </label>
                                    <input className={Styles.field} type='text' defaultValue="0534886574" name='melliCodeField' onChange={this.handleChange} />

                                    <button type='button' className={Styles.btn} onClick={this.submit} name="searchByMelliCodeBtn">جستجو</button>

                                </div>
                            </If>
                            <If condition={this.props.viewState.searchPropertyByPostCode}>
                                <div>
                                    <label className={Styles.label} htmlFor="postcodeField">کد پستی </label>
                                    <input className={Styles.field} type='text' defaultValue="3813783188" name='postcodeField' onChange={this.handleChange} />

                                    <button type='button' className={Styles.btn} onClick={this.submit} name="searchBypostcodeBtn">جستجو</button>

                                </div>
                            </If>
                            <If condition={this.props.viewState.isLoadingSearchResult}>
                                <img src={img} className={Styles.img} /><label className={Styles.label} >اتصال به پایگاه داده ثبت احوال ایران </label>
                            </If>
                            <If condition={this.props.viewState.isLoadingSearchResultPost}>
                                <img src={img} className={Styles.img} /><label className={Styles.label} >اتصال به پایگاه داده پست ایران </label>
                            </If>
                        </div>
                    </If>

                    <If condition={this.hasUserAddedData()}>
                        <div className={Styles.addedData}>
                            <p className={Styles.explanation}>
                                Data added in this way is not saved or made visible to others unless you explicitly share
                                it by using the Share panel.
                            </p>
                            <ul className={Styles.dataCatalog}>
                                <DataCatalogGroup group={this.props.terria.catalog.userAddedDataGroup}
                                    viewState={this.props.viewState} />
                            </ul>
                            <button type='button'
                                onClick={this.onAddMoreDataButtonClicked}
                                className={Styles.btnAddMoreData}>
                                Add more data
                            </button>
                        </div>
                    </If>
                </div>
                <DataPreview terria={this.props.terria}
                    viewState={this.props.viewState}
                    previewed={this.props.viewState.userDataPreviewedItem}
                />
            </div>
        );
    },
});

module.exports = MyDataTab;
